### PR TITLE
Add source runner

### PR DESCRIPTION
**Pull request by jasonLaster**
Refs #1280;

This adds a simple interface for seeing the AST output of esprima. The editor is **code mirror** and the json view is **json editor**.  All of the assets are pulled down via **cdnjs** and the setup is minimal.

Next Steps. I'll add checkboxes for esprima options and a viewer for tokens and raw output. 

Benefit. This makes it easy to test a specific use case while you're debugging locally with chrome devtools. The breakpoint photo below is the main example.

### CODE REVIEW

File: test/runner.html [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        <script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.6.0/codemirror.js"></script>
+        <script src="https://cdnjs.cloudflare.com/ajax/libs/codemirror/5.6.0/mode/javascript/javascript.js"></script>
+
+        <style>
+          .CodeMirror {
+            border-bottom: 1px solid #ccc;
+          }
+
+          #output-tree {
+            height: calc(100% - 110px);
```
Comment (jasonLaster) [inline test/runner.html]: Responsive design :) Full width, Full Height. Just needs one magic number *110*.

**Comment by ariya** [comment]
Seems good to me. One question: as I try it, the code is automatically parsed for the first time and hence the tree just shows (empty object). Am I missing something?

**Comment by jasonLaster** [comment]
@ariya the code is parsed on ENTER. I'll change that to be every keypress.

**Comment by ariya** [comment]
:+1:

**Pull request closed without merging**